0048-schema-update-process/README.md

```diff
@@ -0,0 +1,422 @@
+
+A new field on all future schemas called `.build.apiVersion` will be introduced to track which apiVersion was used at build time. This field will be used to determine which version of the package definition will be printed to the user during `zarf package inspect definition` and the interactive prompts of `zarf package deploy|remove`. 
+
+### Conversions
+
+Zarf will need to handle two use cases for conversions. The first is library convert functions. These functions will provide a path for existing packages to call packager functions after they change accept v1beta1 objects. The second is `zarf dev upgrade-schema` which will provide a simple way for users to convert their zarf.yaml files from one schema version to the next.
+
+#### Library Conversions
+
+There will be a new package and type `generic.ZarfPackage` that is used solely for conversions. Rather than having functions which convert v1alpha1 to v1beta1, functions will instead convert v1alpha1 to the generic Zarf package type then convert the generic Zarf package type to v1beta1. This means Zarf only needs N conversion functions (N API versions) rather than N² conversions between every pair of versions. 
+
+The [api](https://github.com/zarf-dev/zarf/tree/main/src/api) package will be structured as below:
+```bash
+├── generic
+│   └── package.go
+├── v1alpha1
+│   ├── convert.go
+│   ├── package.go
+│   ├── ...
+├── v1beta1
+│   ├── convert.go
+│   ├── package.go
+│   ├── ...
+```
+The `convert.go` file in each versioned package will contain public functions for converting to and from the generic type. These functions will look like below:
+
+`func ConvertToGeneric(in ZarfPackage) generic.ZarfPackage`
+
+`func ConvertFromGeneric(in generic.ZarfPackage) ZarfPackage`
+
+##### Converting 1:1 Replacements
+If a field is renamed with a 1:1 replacement, then Zarf will automatically convert the field to its replacement. For example, if a field called `noWait` was changed to `wait` then the value of the field will flip during conversion
+
+##### Converting Removed Fields
+
+When Zarf internally converts an older schema version to a newer schema version (for example, while deploying a v1alpha1 package), it must always convert to the latest schema version without data loss. To achieve this, fields that were removed from earlier schema versions are preserved as private fields in newer objects. These private fields are kept out of the new schema.
+
+In the below example, `dataInjections` is set as a private field on Zarf components so that it can be set during conversions between v1alpha1 and v1beta1. It will not be included in the v1beta1 schema, and since Zarf validates against the schema on create, users will be unable to create v1beta1 packages with `dataInjections` set. Likewise, since `dataInjections` is a private field, SDK users will not be able to set it directly. 
```

Comment:
In your example you have `DataInjections` available in `v1alpha1`, and now the question is: the new `dataInjections` where it will be available? Generic? `v1beta1`? Both? Can you describe this more, please?